docs/checkit.js

```diff
@@ -130,6 +130,102 @@ Calendar.prototype.generateMonthObjects = function (startDate, endDate) {
 
 };
 
+// calendarAnalyzer takes a calendar object and can analyze facts about it.
+
+var CalendarAnalyzer = function (calendarState) {
+    "use strict";
+    this.calState = calendarState;
+};
+
+CalendarAnalyzer.prototype.getNumberOfChecked = function () {
+    // Returns the number of checked days in a calendar
+    "use strict";
+    return Object.keys(this.calState.checkedDays).length;
+};
+
+CalendarAnalyzer.prototype.getTotalCalendarDays = function () {
+    // Returns the total number of active days in the calendar
+    "use strict";
+    var endDate = moment(this.calState.endDateString, "YYYYMMDD"),
+        startDate = moment(this.calState.startDateString, "YYYYMMDD"),
+        totalDays = endDate.diff(startDate, 'days');
+    return totalDays;
+};
+
+CalendarAnalyzer.prototype.getNumberOfUnchecked = function () {
+    // Returns number of unchecked days in a calendar
+    "use strict";
+    return this.getTotalCalendarDays() - this.getNumberOfChecked();
+};
+
+CalendarAnalyzer.prototype.getCheckedDaysStreak = function () {
+    // Returns the longest streak of checked days
+    "use strict";
+    
+    var currentStreak = 0,
+        checkedDaysArray = [],
+        lastDate,
+        bestStreak = 0,
+        i,
+        currentDay,
+        previousDay,
+        dayDiff;
+    
+    // place checkedDays object's keys into array and sort the array.
+    checkedDaysArray = Object.keys(this.calState.checkedDays);
+    checkedDaysArray.sort();
+
+    lastDate = checkedDaysArray[0];
+    
+    for (i = 1; i < checkedDaysArray.length; i++) {
+        // if the next element is the day after lastDate, add 1 to currentStreak
+        currentDay = moment(checkedDaysArray[i], "YYYYMMDD");
+        previousDay = moment(checkedDaysArray[i - 1], "YYYYMMDD");
+        
+        dayDiff = currentDay.diff(previousDay, 'days');
+
+        currentStreak++;
+        if (dayDiff !== 1) {
+            if (bestStreak < currentStreak) {
+                bestStreak = currentStreak;
+            }
+            // reset currentStreak
+            currentStreak = 0;
+        }
+    }
+
+    return bestStreak;
+};
+
+CalendarAnalyzer.prototype.getUncheckedDaysStreak = function () {
+    // Returns the longest streak of unchecked days
+    "use strict";
+    // TODO
+    // Go through all the dates between startDate and endDate, if they
+    // are not in checkedDays add them to an uncheckedDays array
+    // then find the bestStreak in that uncheckedDays array. 
```

Comment:
Reinterpreting a new problem as a problem you already know to solve is a good strategy. In this particular case I believe there's an easier way. The largest gap between any two checked days (minus one) is all you need I think.